---
title: "Single Cell ATAC-seq analysis tutorial (PBMC3K Example based R Programming Language-Signac)"
permalink: /posts/blog-sc-atac-seq/
date: 2023-10-01
excerpt: 'Single Cell ATAC-seq analysis tutorial. <br/>
<img src="/images/Signac-8.png" style="width:600px; height:250px;" align="center">'
tags:
  - Bioinformatics
---


Background
======
Single-cell ATAC-seq (scATAC-seq) is a powerful technique that enables the study of chromatin accessibility at the single-cell level, providing insights into the regulation of gene expression, cellular heterogeneity, and tissue-specific regulatory mechanisms. It combines the advantages of single-cell RNA sequencing (scRNA-seq) with the ability to profile chromatin dynamics, specifically identifying open chromatin regions that are accessible to transcription factors, enhancers, and other regulatory proteins.<br>

The fundamental principle behind scATAC-seq is the use of the transposase enzyme to insert sequencing adapters into open chromatin regions, which can then be analyzed to identify regions of the genome that are active or primed for transcription. Unlike bulk ATAC-seq, which averages data from millions of cells, scATAC-seq enables the assessment of chromatin accessibility in individual cells, allowing for the identification of cell-type-specific regulatory landscapes.<br>

This technique has emerged as a critical tool for understanding the molecular basis of development, immune responses, disease progression, and cell plasticity. It is particularly useful in elucidating the regulatory mechanisms in complex tissues, such as tumors, immune cells, and the brain, where cellular diversity and dynamic changes in chromatin accessibility play pivotal roles in disease pathogenesis and treatment responses.<br>

In recent years, improvements in both computational methods and experimental protocols have made scATAC-seq more accessible and scalable. Tools such as Signac, an R-based package for single-cell analysis, enable efficient analysis and visualization of ATAC-seq data, making it easier to derive meaningful insights from large-scale datasets.<br>

Overall, scATAC-seq is a transformative technique in the field of genomics, providing a more nuanced understanding of chromatin accessibility and gene regulation at the single-cell level. It holds immense potential for uncovering cellular heterogeneity, uncovering new biomarkers, and developing more precise therapeutic strategies for a variety of diseases.<br>

Method
======

### Step 0. Import Seurat package
First of all, loading Seurat package
```R
library(Signac)
library(Seurat)
library(EnsDb.Hsapiens.v75)
library(ggplot2)
library(patchwork)
```
 
 ### Step 1. Pre-processing workflow
 When pre-processing chromatin data, Signac uses information from two related input files, both of which can be created using CellRanger. And the need the file (Raw data,  Metadata, fragments file, index). We start by creating a Seurat object using the peak/cell matrix and cell metadata generated by cellranger-atac, and store the path to the fragment file on disk in the Seurat object:
 ```R
counts <- Read10X_h5(filename = "C:/Users/zengmin/Desktop/Lab/Sequence/SC-ATAC-seq/PBMC-Signac/atac_v1_pbmc_10k_filtered_peak_bc_matrix.h5")
metadata <- read.csv(
  file = "C:/Users/zengmin/Desktop/Lab/Sequence/SC-ATAC-seq/PBMC-Signac/atac_v1_pbmc_10k_singlecell.csv",
  header = TRUE,
  row.names = 1
)

chrom_assay <- CreateChromatinAssay(
  counts = counts,
  sep = c(":", "-"),
  fragments = 'C:/Users/zengmin/Desktop/Lab/Sequence/SC-ATAC-seq/PBMC-Signac/atac_v1_pbmc_10k_fragments.tsv.gz',
  min.cells = 10,
  min.features = 200
)

pbmc <- CreateSeuratObject(
  counts = chrom_assay,
  assay = "peaks",
  meta.data = metadata
)
```
We can also add gene annotations to the pbmc object for the human genome. This will allow downstream functions to pull the gene annotation information directly from the object.
```R
# extract gene annotations from EnsDb
annotations <- GetGRangesFromEnsDb(ensdb = EnsDb.Hsapiens.v75)
# change to UCSC style since the data was mapped to hg19
seqlevels(annotations) <- paste0('chr', seqlevels(annotations))
genome(annotations) <- "hg19"
# add the gene information to the object
Annotation(pbmc) <- annotations
```


### Step 2. Quality control
We can now compute some QC metrics for the scATAC-seq experiment. We currently suggest the following metrics below to assess data quality. As with scRNA-seq, the expected range of values for these parameters will vary depending on your biological system, cell viability, and other factors. (Nucleosome banding pattern, Transcriptional start site (TSS) enrichment score, Total number of fragments in peaks, Fraction of fragments in peaks)
```R
# compute nucleosome signal score per cell
pbmc <- NucleosomeSignal(object = pbmc)
# compute TSS enrichment score per cell
pbmc <- TSSEnrichment(object = pbmc, fast = FALSE)
# add blacklist ratio and fraction of reads in peaks
pbmc$pct_reads_in_peaks <- pbmc$peak_region_fragments / pbmc$passed_filters * 100
pbmc$blacklist_ratio <- pbmc$blacklist_region_fragments / pbmc$peak_region_fragments
DensityScatter(pbmc, x = 'nCount_peaks', y = 'TSS.enrichment', log_x = TRUE, quantiles = TRUE)
```
Density<img src="/images/Signac-1.png"><br/>
We can inspect the TSS enrichment scores by grouping the cells based on the score and plotting the accessibility signal over all TSS sites.
```R
pbmc$high.tss <- ifelse(pbmc$TSS.enrichment > 3, 'High', 'Low')
TSSPlot(pbmc, group.by = 'high.tss') + NoLegend()
```
**TSS enrichment scores** <img src="/images/Signac-2.png"><br/>
We can also look at the fragment length periodicity for all the cells, and group by cells with high or low nucleosomal signal strength.
```R
pbmc$nucleosome_group <- ifelse(pbmc$nucleosome_signal > 4, 'NS > 4', 'NS < 4')
FragmentHistogram(object = pbmc, group.by = 'nucleosome_group')
```
**TSS enrichment scores** <img src="/images/Signac-3.png"><br/>
We can plot the distribution of each QC metric separately using a violin plot:
```R
VlnPlot(
  object = pbmc,
  features = c('nCount_peaks', 'TSS.enrichment', 'blacklist_ratio', 'nucleosome_signal', 'pct_reads_in_peaks'),
  pt.size = 0.1,
  ncol = 5
)
```
**TSS enrichment scores** <img src="/images/Signac-4.png"><br/>
Finally we remove cells that are outliers for these QC metrics.
```R
pbmc <- subset(
  x = pbmc,
  subset = nCount_peaks > 3000 &
    nCount_peaks < 30000 &
    pct_reads_in_peaks > 15 &
    blacklist_ratio < 0.05 &
    nucleosome_signal < 4 &
    TSS.enrichment > 3
)
pbmc
```

### Step 3. Normalization and linear dimensional reduction
```R
pbmc <- RunTFIDF(pbmc)
pbmc <- FindTopFeatures(pbmc, min.cutoff = 'q0')
pbmc <- RunSVD(pbmc)
DepthCor(pbmc)
```
**Correlation** <img src="/images/Signac-5.png"><br/>

### Step 4. Non-linear dimension reduction and clustering
Now that the cells are embedded in a low-dimensional space we can use methods commonly applied for the analysis of scRNA-seq data to perform graph-based clustering and non-linear dimension reduction for visualization. The functions RunUMAP(), FindNeighbors(), and FindClusters() all come from the Seurat package.

```R
pbmc <- RunUMAP(object = pbmc, reduction = 'lsi', dims = 2:30)
pbmc <- FindNeighbors(object = pbmc, reduction = 'lsi', dims = 2:30)
pbmc <- FindClusters(object = pbmc, verbose = FALSE, algorithm = 3)
DimPlot(object = pbmc, label = TRUE) + NoLegend()
```
**DimPlot Figure** <img src="/images/Signac-6.png"><br/>

### Step 5. Create a gene activity matrix
```R
gene.activities <- GeneActivity(pbmc)
# add the gene activity matrix to the Seurat object as a new assay and normalize it
pbmc[['RNA']] <- CreateAssayObject(counts = gene.activities)
pbmc <- NormalizeData(
  object = pbmc,
  assay = 'RNA',
  normalization.method = 'LogNormalize',
  scale.factor = median(pbmc$nCount_RNA)
)
DefaultAssay(pbmc) <- 'RNA'

FeaturePlot(
  object = pbmc,
  features = c('MS4A1', 'CD3D', 'LEF1', 'NKG7', 'TREM1', 'LYZ'),
  pt.size = 0.1,
  max.cutoff = 'q95',
  ncol = 3
)
```
**FeaturePlot Figure** <img src="/images/Signac-7.png"><br/>


### Step 6. Integrating with scRNA-seq data
To help interpret the scATAC-seq data, we can classify cells based on an scRNA-seq experiment from the same biological system (human PBMC). We utilize methods for cross-modality integration and label transfer. We aim to identify shared correlation patterns in the gene activity matrix and scRNA-seq dataset to identify matched biological states across the two modalities. This procedure returns a classification score for each cell for each scRNA-seq-defined cluster label.

```R
# Load the pre-processed scRNA-seq data for PBMCs
pbmc_rna <- readRDS("../vignette_data/pbmc_10k_v3.rds")
pbmc_rna <- UpdateSeuratObject(pbmc_rna)
transfer.anchors <- FindTransferAnchors(
  reference = pbmc_rna,
  query = pbmc,
  reduction = 'cca'
)
predicted.labels <- TransferData(
  anchorset = transfer.anchors,
  refdata = pbmc_rna$celltype,
  weight.reduction = pbmc[['lsi']],
  dims = 2:30
)
plot1 <- DimPlot(
  object = pbmc_rna,
  group.by = 'celltype',
  label = TRUE,
  repel = TRUE) + NoLegend() + ggtitle('scRNA-seq')

plot2 <- DimPlot(
  object = pbmc,
  group.by = 'predicted.id',
  label = TRUE,
  repel = TRUE) + NoLegend() + ggtitle('scATAC-seq')

plot1 + plot2
# replace each label with its most likely prediction
for(i in levels(pbmc)) {
  cells_to_reid <- WhichCells(pbmc, idents = i)
  newid <- names(which.max(table(pbmc$predicted.id[cells_to_reid])))
  Idents(pbmc, cells = cells_to_reid) <- newid
}
```
**sc ATAC seq-sc RNA seq Figure** <img src="/images/Signac-8.png"><br/>

### Step 7. Find differentially accessible peaks between cell types
To find differentially accessible regions between clusters of cells, we can perform a differential accessibility (DA) test. A simple approach is to perform a Wilcoxon rank sum test, and the presto package has implemented an extremely fast Wilcoxon test that can be run on a Seurat object.
```R
# change back to working with peaks instead of gene activities
DefaultAssay(pbmc) <- 'peaks'

da_peaks <- FindMarkers(
  object = pbmc,
  ident.1 = "CD4 Naive",
  ident.2 = "CD14+ Monocytes",
  test.use = 'LR',
  latent.vars = 'nCount_peaks'
)
head(da_peaks)
plot1 <- VlnPlot(
  object = pbmc,
  features = rownames(da_peaks)[1],
  pt.size = 0.1,
  idents = c("CD4 Naive","CD14+ Monocytes")
)
plot2 <- FeaturePlot(
  object = pbmc,
  features = rownames(da_peaks)[1],
  pt.size = 0.1
)

plot1 | plot2
```
**VlnPlot/FeaturePlot** <img src="/images/Signac-9.png"><br/>

Another way to find DA regions between two groups of cells is to look at the fold change accessibility between two groups of cells. This can be much faster than running more sophisticated DA tests, but is not able to account for latent variables such as differences in the total sequencing depth between cells, and does not perform any statistical test. However, this can still be a useful way to quickly explore data, and can be performed using the FoldChange() function in Seurat.
```R
fc <- FoldChange(pbmc, ident.1 = "CD4 Naive", ident.2 = "CD14+ Monocytes")
# order by fold change
fc <- fc[order(fc$avg_log2FC, decreasing = TRUE), ]
head(fc)
open_cd4naive <- rownames(da_peaks[da_peaks$avg_log2FC > 3, ])
open_cd14mono <- rownames(da_peaks[da_peaks$avg_log2FC < -3, ])

closest_genes_cd4naive <- ClosestFeature(pbmc, regions = open_cd4naive)
closest_genes_cd14mono <- ClosestFeature(pbmc, regions = open_cd14mono)
head(closest_genes_cd4naive)
head(closest_genes_cd14mono)
```

### Step 8. Plotting genomic regions
We can plot the frequency of Tn5 integration across regions of the genome for cells grouped by cluster, cell type, or any other metadata stored in the object for any genomic region using the CoveragePlot() function. 
```R
# set plotting order
levels(pbmc) <- c("CD4 Naive","CD4 Memory","CD8 Naive","CD8 effector","Double negative T cell","NK dim", "NK bright", "pre-B cell",'B cell progenitor',"pDC","CD14+ Monocytes",'CD16+ Monocytes')

CoveragePlot(
  object = pbmc,
  region = rownames(da_peaks)[1],
  extend.upstream = 40000,
  extend.downstream = 20000
)
```
**CoveragePlot1** <img src="/images/Signac-10.png"><br/>
**CoveragePlot2** <img src="/images/Signac-11.png"><br/>
