---
title: "CellChat for Single Cell RNA-seq"
permalink: /posts/2015-08-14-blog-post-10/
date: 2025-01-12
excerpt: 'CellChat for Gut Immune Cell. <br/><img src="/images/GSE116093-output.png">'
tags:
  - Bioinformatics
---

Background
======


Results
======
**UMAP**<br/><img src="/images/Bolg-Il18/il18-6.jpg"><br/>
**Database**<br/><img src="/images/blog_cellchat/database.png"><br/>
**All Cellchat**<br/><img src="/images/blog_cellchat/ALL.png"><br/>
**Mast Cellchat**<br/><img src="/images/blog_cellchat/mast.png"><br/>
**CXCL signaling pathway network**<br/><img src="/images/blog_cellchat/cxcl.png"><br/>
**CXCL signaling pathway network2**<br/><img src="/images/blog_cellchat/cxcl2.png"><br/>
**Cxcl4-Cxcr3**<br/><img src="/images/blog_cellchat/Cxcl4_Cxcr3.png"><br/>
**Dotplot**<br/><img src="/images/blog_cellchat/dot.png"><br/>
**VlnPlot**<br/><img src="/images/blog_cellchat/vln.png"><br/>


Method
======

 ### Step 0. Download SRA file and convert into Fastq file (based on SRA-Explorer)
 ```R
# install R package
devtools::install_github("jinworks/CellChat")
install.packages('NMF')
devtools::install_github("jokergoo/circlize")
devtools::install_github("jokergoo/ComplexHeatmap")

# load Rdata
sisc <- readRDS("/data1/zengmin/SC-RNA-seq/GSE124880-SI-CD45-Xuheping/R_Result/deletecell.RData")

# load R package and basic setting
setwd("/data1/zengmin/SC-RNA-seq/GSE124880-SI-CD45-Xuheping/R_Result/Cellchat/")
library(CellChat)
library(patchwork)
options(stringsAsFactors = FALSE)
options(future.globals.maxSize = 50 * 1024^3)

# Starting from a Seurat object
meta = sisc@meta.data
cellchat <- createCellChat(object = sisc, group.by = "celltype", meta = meta)

# Set the ligand-receptor interaction database
CellChatDB <- CellChatDB.mouse # use CellChatDB.mouse if running on mouse data
showDatabaseCategory(CellChatDB)

# use a subset of CellChatDB for cell-cell communication analysis
CellChatDB.use <- subsetDB(CellChatDB)
cellchat@DB <- CellChatDB.use

# Preprocessing the expression data for cell-cell communication analysis
cellchat <- subsetData(cellchat) # This step is necessary even if using the whole database
future::plan("multisession", workers = 32) # do parallel
cellchat <- identifyOverExpressedGenes(cellchat)
cellchat <- identifyOverExpressedInteractions(cellchat)

# Compute the communication probability and infer cellular communication network
idents <- cellchat@idents
idents <- droplevels(idents)
cellchat@idents <- idents
cellchat <- computeCommunProb(cellchat, type = "triMean")

# Infer the cell-cell communication at a signaling pathway level
cellchat <- computeCommunProbPathway(cellchat)

# Calculate the aggregated cell-cell communication network
cellchat <- aggregateNet(cellchat)

# visualization
groupSize <- as.numeric(table(cellchat@idents))
par(mfrow = c(1,2), xpd=TRUE)
netVisual_circle(cellchat@net$count, vertex.weight = groupSize, weight.scale = T, label.edge= F, title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight, vertex.weight = groupSize, weight.scale = T, label.edge= F, title.name = "Interaction weights/strength")

mat <- cellchat@net$weight
par(mfrow = c(3,4), xpd=TRUE)
for (i in 1:nrow(mat)) {
  mat2 <- matrix(0, nrow = nrow(mat), ncol = ncol(mat), dimnames = dimnames(mat))
  mat2[i, ] <- mat[i, ]
  netVisual_circle(mat2, vertex.weight = groupSize, weight.scale = T, edge.weight.max = max(mat), title.name = rownames(mat)[i])
  
# Visualize each signaling pathway using Hierarchy plot, Circle plot or Chord diagram
pathways.show <- c("CXCL") 
# Hierarchy plot
# Here we define `vertex.receive` so that the left portion of the hierarchy plot shows signaling to fibroblast and the right portion shows signaling to immune cells 
vertex.receiver = seq(1,4) # a numeric vector. 
netVisual_aggregate(cellchat, signaling = pathways.show,  vertex.receiver = vertex.receiver)
# Circle plot
par(mfrow=c(1,1))
netVisual_aggregate(cellchat, signaling = pathways.show, layout = "circle")
}

# Chord diagram
par(mfrow=c(1,1))
netVisual_aggregate(cellchat, signaling = pathways.show, layout = "chord")

# Chord diagram
group.cellType <- c(rep("FIB", 4), rep("DC", 4), rep("TC", 4)) # grouping cell clusters into fibroblast, DC and TC cells
names(group.cellType) <- levels(cellchat@idents)
netVisual_chord_cell(cellchat, signaling = pathways.show, group = group.cellType, title.name = paste0(pathways.show, " signaling network"))
```



