---
title: "Using QIIME 2 to Analyze 16S rRNA Data from Parkinson's Disease Humanized Mice"
permalink: /posts/2015-08-14-blog-post-40/
date: 2025-06-16
excerpt: 'This tutorial demonstrates a typical QIIME 2 pipeline using 16S rRNA amplicon data derived from humanized mouse fecal samples in a Parkinson’s disease study.<br/>
<img src="/images/Qiime2/level_4_bars.png" style="width:600px; height:600px;">'


tags:
  - Bioinformatics
---

Background
======
The mammalian gastrointestinal tract is home to a diverse and highly dynamic microbial ecosystem that plays a fundamental role in shaping host immune function and maintaining intestinal homeostasis. Disruption of this complex microbial community—commonly referred to as **gut dysbiosis**—has been increasingly implicated in the pathogenesis of various immune-mediated diseases, most notably **inflammatory bowel disease (IBD)**. Both Crohn’s disease and ulcerative colitis, the two major forms of IBD, exhibit marked alterations in gut microbial composition and diversity, often characterized by reduced microbial richness and expansion of pro-inflammatory taxa.<br/>

Over the past decade, high-throughput sequencing of 16S rRNA genes has become a cornerstone for profiling gut microbial communities. Among the available computational tools, **QIIME 2 (Quantitative Insights Into Microbial Ecology 2)** stands out as a reproducible, extensible, and scalable platform for microbiome bioinformatics. It provides an end-to-end framework for raw data processing, quality control, taxonomic classification, phylogenetic tree construction, diversity analysis, and differential abundance testing.<br/>

This tutorial leverages QIIME 2 to analyze 16S rRNA gene amplicon sequencing data from a **gnotobiotic mouse model**, transplanted with fecal samples from human donors. While the dataset used here stems from a Parkinson’s disease (PD) study, the analytical pipeline is fully applicable to IBD and other immune-related conditions. Indeed, mouse models of colitis, graft-versus-host disease, and food allergy have all benefited from similar microbiome transplantation approaches to assess the immunomodulatory effects of gut microbes.<br/>

By applying QIIME 2, we aim to:

Evaluate microbial diversity (alpha and beta diversity) in different experimental groups.<br/>
Identify shifts in taxonomic composition associated with disease phenotypes or immune perturbations.<br/>
Infer ecological or functional patterns that may contribute to intestinal inflammation or immune dysregulation.<br/>

Through this pipeline, users will gain hands-on experience with QIIME 2’s core functionalities, including:<br/>


Result
======
**Relative Abundance of Microbial Taxa at Taxonomic Level 4 Across Samples**<br/><img src="/images/Qiime2/level_4_bars.png"><br/><br/>
**Comparison of Alpha Diversity Metrics Across Sample Groups**<br/><img src="/images/Qiime2/alpha_compare.png"><br/><br/>
<br/><img src="/images/Qiime2/alpha_compare2.png"><br/><br/>



```
(base) zengmin@lab:~$ ls -R /data1/zengmin/Microbio/mouse_tutorial/
/data1/zengmin/Microbio/mouse_tutorial/:
alpha_rarefaction_curves.qzv  core-metrics-results              dada2_rep_set.qzv   demultiplexed_seqs.zip                manifest.tsv     ref_tax.qza                taxonomy.qzv
ancombc_donor_genotype.qza    da_barplot_donor_genotype.qzv     dada2_stats.qza     demux_seqs.qza                        metadata.qzv     sample-classifier-results  tree_placements.qza
ancombc_donor.qza             da_barplot_donor.qzv              dada2_stats.qzv     from_first_unifrac_lme.qzv            metadata.tsv     sepp-refs-gg-13-8.qza      tree.qza
ancombc_genotype.qza          da_barplot_genotype.qzv           dada2_table.qza     from_first_unifrac.qza                multi-taxonomy   table_2k_abund.qza
animal_distal_gut.qza         dada2_rep_set_multi_taxonomy.qzv  dada2_table.qzv     from_first_unifrac_vol.qzv            pc_vol.qzv       table_2k.qza
bespoke.qza                   dada2_rep_set.qza                 demultiplexed_seqs  gg-13-8-99-515-806-nb-classifier.qza  ref_seqs_v4.qza  taxa_barplot.qzv

/data1/zengmin/Microbio/mouse_tutorial/core-metrics-results:
bray_curtis_distance_matrix.qza  faith_pd_vector.qza          observed_features_vector.qza              unweighted_unifrac_distance_matrix.qza       weighted-unifrac-cage-significance.qzv
bray_curtis_emperor.qzv          faiths_pd_statistics.qzv     rarefied_table.qza                        unweighted-unifrac-donor-significance.qzv    weighted_unifrac_distance_matrix.qza
bray_curtis_pcoa_results.qza     jaccard_distance_matrix.qza  shannon_vector.qza                        unweighted_unifrac_emperor.qzv               weighted-unifrac-donor-significance.qzv
evenness_statistics.qzv          jaccard_emperor.qzv          unweighted_adonis.qzv                     unweighted_unifrac_pcoa_results.qza          weighted_unifrac_emperor.qzv
evenness_vector.qza              jaccard_pcoa_results.qza     unweighted-unifrac-cage-significance.qzv  weighted-unifrac-cage-significance_disp.qzv  weighted_unifrac_pcoa_results.qza

/data1/zengmin/Microbio/mouse_tutorial/demultiplexed_seqs:
10483.recip.220.WT.OB1.D7_30_L001_R1_001.fastq.gz    10483.recip.412.ASO.HC2.D7_24_L001_R1_001.fastq.gz   10483.recip.461.ASO.HC3.D21_11_L001_R1_001.fastq.gz  10483.recip.465.ASO.PD3.D7_0_L001_R1_001.fastq.gz
10483.recip.290.ASO.OB2.D1_27_L001_R1_001.fastq.gz   10483.recip.413.WT.HC2.D7_12_L001_R1_001.fastq.gz    10483.recip.461.ASO.HC3.D49_29_L001_R1_001.fastq.gz  10483.recip.466.ASO.PD3.D7_23_L001_R1_001.fastq.gz
10483.recip.389.WT.HC2.D21_1_L001_R1_001.fastq.gz    10483.recip.456.ASO.HC3.D49_28_L001_R1_001.fastq.gz  10483.recip.461.ASO.HC3.D7_20_L001_R1_001.fastq.gz   10483.recip.467.WT.HC3.D49.a_0_L001_R1_001.fastq.gz
10483.recip.391.ASO.PD2.D14_5_L001_R1_001.fastq.gz   10483.recip.458.ASO.HC3.D21_10_L001_R1_001.fastq.gz  10483.recip.462.WT.PD3.D14_18_L001_R1_001.fastq.gz   10483.recip.467.WT.HC3.D49.b_26_L001_R1_001.fastq.gz
10483.recip.391.ASO.PD2.D21_1_L001_R1_001.fastq.gz   10483.recip.458.ASO.HC3.D49_2_L001_R1_001.fastq.gz   10483.recip.462.WT.PD3.D21_4_L001_R1_001.fastq.gz    10483.recip.536.ASO.PD4.D49_1_L001_R1_001.fastq.gz
10483.recip.391.ASO.PD2.D7_15_L001_R1_001.fastq.gz   10483.recip.459.WT.HC3.D14_2_L001_R1_001.fastq.gz    10483.recip.462.WT.PD3.D49_8_L001_R1_001.fastq.gz    10483.recip.537.WT.PD4.D21_45_L001_R1_001.fastq.gz
10483.recip.400.ASO.HC2.D14_32_L001_R1_001.fastq.gz  10483.recip.459.WT.HC3.D21_21_L001_R1_001.fastq.gz   10483.recip.462.WT.PD3.D7_3_L001_R1_001.fastq.gz     10483.recip.538.WT.PD4.D21_4_L001_R1_001.fastq.gz
10483.recip.401.ASO.HC2.D7_22_L001_R1_001.fastq.gz   10483.recip.459.WT.HC3.D49_3_L001_R1_001.fastq.gz    10483.recip.463.WT.PD3.D14_25_L001_R1_001.fastq.gz   10483.recip.539.ASO.PD4.D14_5_L001_R1_001.fastq.gz
10483.recip.403.ASO.PD2.D21_31_L001_R1_001.fastq.gz  10483.recip.460.WT.HC3.D14_11_L001_R1_001.fastq.gz   10483.recip.463.WT.PD3.D21_19_L001_R1_001.fastq.gz   10483.recip.539.ASO.PD4.D7_4_L001_R1_001.fastq.gz
10483.recip.411.ASO.HC2.D14_9_L001_R1_001.fastq.gz   10483.recip.460.WT.HC3.D21_0_L001_R1_001.fastq.gz    10483.recip.463.WT.PD3.D7_3_L001_R1_001.fastq.gz     10483.recip.540.ASO.HC4.D14_2_L001_R1_001.fastq.gz
10483.recip.411.ASO.HC2.D49_6_L001_R1_001.fastq.gz   10483.recip.460.WT.HC3.D49_19_L001_R1_001.fastq.gz   10483.recip.464.WT.PD3.D14_17_L001_R1_001.fastq.gz   10483.recip.540.ASO.HC4.D7_7_L001_R1_001.fastq.gz
10483.recip.412.ASO.HC2.D14_13_L001_R1_001.fastq.gz  10483.recip.460.WT.HC3.D7_14_L001_R1_001.fastq.gz    10483.recip.465.ASO.PD3.D14_16_L001_R1_001.fastq.gz  recip.411.ASO.HC2.D21_0_L001_R1_001.fastq.gz

/data1/zengmin/Microbio/mouse_tutorial/multi-taxonomy:
bespoke_taxonomy.qza  taxonomy.qza

/data1/zengmin/Microbio/mouse_tutorial/sample-classifier-results:
accuracy_results.qzv    filtered-table_100-features.qza  heatmap.qzv        predictions.qza    sample_estimator.qza  training_targets.qza
feature_importance.qza  heatmap_100-features.qzv         model_summary.qzv  probabilities.qza  test_targets.qza
```



Method
======
```python
# Install QIIME 2 environment (Estimated time: 10–20 minutes)
conda env create -n qiime2-amplicon-2024.10 --file https://data.qiime2.org/distro/amplicon/qiime2-amplicon-2024.10-py310-linux-conda.yml


# Activate the QIIME 2 environment and download example data
conda activate qiime2-amplicon-2024.10

cd /data1/zengmin/Microbio
mkdir ./mouse_tutorial
cd ./mouse_tutorial
wget \
  -O "metadata.tsv" \
  "https://data.qiime2.org/2024.10/tutorials/pd-mice/sample_metadata.tsv"

# The sample metadata will be used throughout the tutorial.
# Let us begin by running the first QIIME 2 command to summarize and explore the metadata.
qiime metadata tabulate \
  --m-input-file metadata.tsv \
  --o-visualization metadata.qzv

# Download manifest and sequence files
wget \
  -O "manifest.tsv" \
  "https://data.qiime2.org/2024.10/tutorials/pd-mice/manifest"

wget \
  -O "demultiplexed_seqs.zip" \
  "https://data.qiime2.org/2024.10/tutorials/pd-mice/demultiplexed_seqs.zip"

# Unzip the downloaded sequence data
unzip demultiplexed_seqs.zip

# Perform sequence quality control and generate the feature table
qiime tools import \
  --type "SampleData[SequencesWithQuality]" \
  --input-format SingleEndFastqManifestPhred33V2 \
  --input-path ./manifest.tsv \
  --output-path ./demux_seqs.qza

qiime dada2 denoise-single \
  --i-demultiplexed-seqs ./demux_seqs.qza \
  --p-trunc-len 150 \
  --o-table ./dada2_table.qza \
  --o-representative-sequences ./dada2_rep_set.qza \
  --o-denoising-stats ./dada2_stats.qza

# Visualize the denoising statistics
qiime metadata tabulate \
  --m-input-file ./dada2_stats.qza  \
  --o-visualization ./dada2_stats.qzv

# Summarize the feature table
qiime feature-table summarize \
  --i-table ./dada2_table.qza \
  --m-sample-metadata-file ./metadata.tsv \
  --o-visualization ./dada2_table.qzv

# Generate a phylogenetic tree for downstream diversity analyses
wget \
  -O "sepp-refs-gg-13-8.qza" \
  "https://data.qiime2.org/classifiers/sepp-ref-dbs/sepp-refs-gg-13-8.qza"
qiime fragment-insertion sepp \
  --i-representative-sequences ./dada2_rep_set.qza \
  --i-reference-database sepp-refs-gg-13-8.qza \
  --o-tree ./tree.qza \
  --o-placements ./tree_placements.qza \
  --p-threads 40  # Adjust based on available CPU cores

# Alpha rarefaction analysis to determine appropriate sampling depth
qiime diversity alpha-rarefaction \
  --i-table ./dada2_table.qza \
  --m-metadata-file ./metadata.tsv \
  --o-visualization ./alpha_rarefaction_curves.qzv \
  --p-min-depth 10 \
  --p-max-depth 4250

# Core diversity metrics computation
qiime diversity core-metrics-phylogenetic \
  --i-table ./dada2_table.qza \
  --i-phylogeny ./tree.qza \
  --m-metadata-file ./metadata.tsv \
  --p-sampling-depth 2000 \
  --output-dir ./core-metrics-results

# Alpha diversity statistical significance testing
qiime diversity alpha-group-significance \
  --i-alpha-diversity ./core-metrics-results/faith_pd_vector.qza \
  --m-metadata-file ./metadata.tsv \
  --o-visualization ./core-metrics-results/faiths_pd_statistics.qzv

qiime diversity alpha-group-significance \
 --i-alpha-diversity ./core-metrics-results/evenness_vector.qza \
 --m-metadata-file ./metadata.tsv \
 --o-visualization ./core-metrics-results/evenness_statistics.qzv

# Beta diversity group significance testing
qiime diversity beta-group-significance \
  --i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza \
  --m-metadata-file metadata.tsv \
  --m-metadata-column donor \
  --o-visualization core-metrics-results/unweighted-unifrac-donor-significance.qzv

qiime diversity beta-group-significance \
  --i-distance-matrix core-metrics-results/weighted_unifrac_distance_matrix.qza \
  --m-metadata-file metadata.tsv \
  --m-metadata-column donor \
  --o-visualization core-metrics-results/weighted-unifrac-donor-significance.qzv

qiime diversity beta-group-significance \
  --i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza \
  --m-metadata-file metadata.tsv \
  --m-metadata-column cage_id \
  --o-visualization core-metrics-results/unweighted-unifrac-cage-significance.qzv \
  --p-pairwise

qiime diversity beta-group-significance \
  --i-distance-matrix core-metrics-results/weighted_unifrac_distance_matrix.qza \
  --m-metadata-file metadata.tsv \
  --m-metadata-column cage_id \
  --o-visualization core-metrics-results/weighted-unifrac-cage-significance.qzv \
  --p-pairwise

qiime diversity beta-group-significance \
  --i-distance-matrix core-metrics-results/weighted_unifrac_distance_matrix.qza \
  --m-metadata-file metadata.tsv \
  --m-metadata-column cage_id \
  --o-visualization core-metrics-results/weighted-unifrac-cage-significance_disp.qzv \
  --p-method permdisp

qiime diversity adonis \
  --i-distance-matrix core-metrics-results/unweighted_unifrac_distance_matrix.qza \
  --m-metadata-file metadata.tsv \
  --o-visualization core-metrics-results/unweighted_adonis.qzv \
  --p-formula genotype+donor

# Taxonomic classification
wget \
  -O "gg-13-8-99-515-806-nb-classifier.qza" \
  "https://data.qiime2.org/classifiers/sklearn-1.4.2/greengenes/gg-13-8-99-515-806-nb-classifier.qza"

qiime feature-classifier classify-sklearn \
  --i-reads ./dada2_rep_set.qza \
  --i-classifier ./gg-13-8-99-515-806-nb-classifier.qza \
  --o-classification ./taxonomy.qza

# Visualize the taxonomic annotations
qiime metadata tabulate \
  --m-input-file ./taxonomy.qza \
  --o-visualization ./taxonomy.qzv

qiime feature-table tabulate-seqs \
  --i-data ./dada2_rep_set.qza \
  --o-visualization ./dada2_rep_set.qzv

# Generate taxonomic bar plots
qiime feature-table filter-samples \
  --i-table ./dada2_table.qza \
  --p-min-frequency 2000 \
  --o-filtered-table ./table_2k.qza

qiime taxa barplot \
  --i-table ./table_2k.qza \
  --i-taxonomy ./taxonomy.qza \
  --m-metadata-file ./metadata.tsv \
  --o-visualization ./taxa_barplot.qzv

# Differential abundance analysis using ANCOM-BC
qiime feature-table filter-features \
  --i-table ./table_2k.qza \
  --p-min-frequency 50 \
  --p-min-samples 4 \
  --o-filtered-table ./table_2k_abund.qza

qiime composition ancombc \
  --i-table ./table_2k_abund.qza \
  --m-metadata-file ./metadata.tsv \
  --p-formula 'donor' \
  --o-differentials ./ancombc_donor.qza

qiime composition da-barplot \
  --i-data ./ancombc_donor.qza \
  --p-significance-threshold 0.001 \
  --o-visualization da_barplot_donor.qzv

qiime composition ancombc \
  --i-table ./table_2k_abund.qza \
  --m-metadata-file ./metadata.tsv \
  --p-formula 'genotype' \
  --o-differentials ./ancombc_genotype.qza

qiime composition da-barplot \
  --i-data ./ancombc_genotype.qza \
  --p-significance-threshold 0.001 \
  --o-visualization da_barplot_genotype.qzv

qiime composition ancombc \
  --i-table ./table_2k_abund.qza \
  --m-metadata-file ./metadata.tsv \
  --p-formula 'donor + genotype' \
  --o-differentials ./ancombc_donor_genotype.qza

qiime composition da-barplot \
  --i-data ./ancombc_donor_genotype.qza \
  --p-significance-threshold 0.001 \
  --o-visualization da_barplot_donor_genotype.qzv

# Alternative taxonomy classification with customized reference
wget \
  -O "ref_seqs_v4.qza" \
  "https://data.qiime2.org/2024.10/tutorials/pd-mice/ref_seqs_v4.qza"
wget \
  -O "ref_tax.qza" \
  "https://data.qiime2.org/2024.10/tutorials/pd-mice/ref_tax.qza"
wget \
  -O "animal_distal_gut.qza" \
  "https://data.qiime2.org/2024.10/tutorials/pd-mice/animal_distal_gut.qza"

qiime feature-classifier fit-classifier-naive-bayes \
  --i-reference-reads ./ref_seqs_v4.qza \
  --i-reference-taxonomy ./ref_tax.qza \
  --i-class-weight ./animal_distal_gut.qza \
  --o-classifier ./bespoke.qza

qiime feature-classifier classify-sklearn \
  --i-reads ./dada2_rep_set.qza \
  --i-classifier ./bespoke.qza \
  --o-classification ./bespoke_taxonomy.qza

mkdir ./multi-taxonomy
  mv ./taxonomy.qza ./multi-taxonomy
  mv ./bespoke_taxonomy.qza ./multi-taxonomy

# Combine multiple taxonomy assignments for visualization
qiime feature-table tabulate-seqs \
  --i-data dada2_rep_set.qza \
  --i-taxonomy multi-taxonomy/ \
  --o-visualization dada2_rep_set_multi_taxonomy.qzv

# Animated PCoA visualization to assess temporal dynamics
qiime longitudinal volatility \
  --m-metadata-file ./metadata.tsv \
  --m-metadata-file ./core-metrics-results/unweighted_unifrac_pcoa_results.qza \
  --p-state-column days_post_transplant \
  --p-individual-id-column mouse_id \
  --p-default-group-column 'donor_status' \
  --p-default-metric 'Axis 2' \
  --o-visualization ./pc_vol.qzv

# Longitudinal distance-based analyses
qiime longitudinal first-distances \
  --i-distance-matrix ./core-metrics-results/unweighted_unifrac_distance_matrix.qza \
  --m-metadata-file ./metadata.tsv \
  --p-state-column days_post_transplant \
  --p-individual-id-column mouse_id \
  --p-baseline 7 \
  --o-first-distances ./from_first_unifrac.qza

qiime longitudinal volatility \
  --m-metadata-file ./metadata.tsv \
  --m-metadata-file ./from_first_unifrac.qza \
  --p-state-column days_post_transplant \
  --p-individual-id-column mouse_id \
  --p-default-metric Distance \
  --p-default-group-column 'donor_status' \
  --o-visualization ./from_first_unifrac_vol.qzv

qiime longitudinal linear-mixed-effects \
  --m-metadata-file ./metadata.tsv \
  --m-metadata-file ./from_first_unifrac.qza \
  --p-metric Distance \
  --p-state-column days_post_transplant \
  --p-individual-id-column mouse_id \
  --p-group-columns genotype,donor \
  --o-visualization ./from_first_unifrac_lme.qzv

# Supervised classification to predict sample features
qiime sample-classifier classify-samples \
  --i-table ./dada2_table.qza \
  --m-metadata-file ./metadata.tsv \
  --m-metadata-column genotype_and_donor_status \
  --p-random-state 666 \
  --p-n-jobs 1 \
  --output-dir ./sample-classifier-results/

qiime sample-classifier heatmap \
  --i-table ./dada2_table.qza \
  --i-importance ./sample-classifier-results/feature_importance.qza \
  --m-sample-metadata-file ./metadata.tsv \
  --m-sample-metadata-column genotype_and_donor_status \
  --p-group-samples \
  --p-feature-count 100 \
  --o-heatmap ./sample-classifier-results/heatmap_100-features.qzv \
  --o-filtered-table ./sample-classifier-results/filtered-table_100-features.qza

```
