---
title: "Single-Cell ATAC Sequencing (scATAC-seq)"
permalink: /posts/2015-08-14-blog-post-13/
date: 2024-10-20
excerpt: 'SnapATAC2 for PBMC by python jupyter lab <br/>
<img src="img src="/images/Bolg-snapatac2/Feature.png" style="width:800px; height:450px;">'


tags:
  - Bioinformatics
---

Background
======
Single-cell Assay for Transposase-Accessible Chromatin using sequencing (scATAC-seq) has emerged as a powerful technique for profiling chromatin accessibility at the single-cell level, enabling the study of regulatory landscapes across heterogeneous cell populations. By leveraging the Tn5 transposase to preferentially insert sequencing adapters into accessible chromatin regions, scATAC-seq provides insights into the epigenetic regulation of gene expression, transcription factor binding, and cell-type-specific regulatory programs. Compared to bulk ATAC-seq, scATAC-seq allows for the resolution of cellular heterogeneity, making it particularly valuable for studying complex tissues, developmental processes, and disease mechanisms. <br/>

However, the inherent sparsity and high-dimensional nature of scATAC-seq data pose significant challenges for computational analysis, necessitating specialized tools for preprocessing, clustering, and interpretation. SnapATAC2, an improved version of the original SnapATAC framework, addresses these challenges by offering a scalable and efficient pipeline for analyzing large-scale single-cell chromatin accessibility data. It integrates advanced computational techniques for dimensionality reduction, peak calling, and cell clustering, facilitating the identification of distinct cellular states and their regulatory elements. With its optimized performance and flexibility, SnapATAC2 provides a robust solution for extracting meaningful biological insights from scATAC-seq datasets, thereby advancing our understanding of gene regulatory networks at the single-cell level.


Result
======
**QC1**<br/><img src="/images/Bolg-snapatac2/QC1.png"><br/>
**QC1**<br/><img src="/images/Bolg-snapatac2/QC1.png"><br/>
**UMAP**<br/><img src="/images/Bolg-snapatac2/UMAP.png"><br/>
**Feature mark gene**<br/><img src="/images/Bolg-snapatac2/Feature.png"><br/>



Method
======

 ###  Jupyter Lab setting
 ```python
#Jupyter lab setting
# base_config
jupyter lab --generate-config

# password
jupyter lab password

# modify config (X)
nano ~/.jupyter/jupyter_lab_config.py

# IP
c.ServerApp.ip = '0.0.0.0'

# open_browser
c.ServerApp.open_browser = False

# port
c.ServerApp.port = 8888

# password
c.ServerApp.password = 'password'

# remote
c.ServerApp.allow_remote_access = True

jupyter lab
http://<IP>:8888

# new environment
conda create -n test python=3.10.9
conda activate test
conda install ipykernel
ipython kernel install --user --name=test
conda install -c conda-forge nodejs
jupyter lab

# or base (not recommend)
conda install nb_conda_kernels
```


### Snapatac2

```python
import snapatac2 as snap
fragment_file = snap.datasets.pbmc5k()
fragment_file

%%time
data = snap.pp.import_fragments(
    fragment_file,
    chrom_sizes=snap.genome.hg38,
    file="pbmc.h5ad",  # Optional
    sorted_by_barcode=False,
)
data

snap.pl.frag_size_distr(data, interactive=False)

fig = snap.pl.frag_size_distr(data, show=False)
fig.update_yaxes(type="log")
fig.show()

%%time
snap.metrics.tsse(data, snap.genome.hg38)

snap.pl.tsse(data, interactive=False)

%%time
snap.pp.filter_cells(data, min_counts=5000, min_tsse=10, max_counts=100000)
data

%%time
snap.pp.add_tile_matrix(data)

snap.pp.select_features(data, n_features=250000)

%%time
snap.pp.scrublet(data)

snap.pp.filter_doublets(data)
data

%%time
snap.tl.spectral(data)

pip install umap-learn
%%time
snap.tl.umap(data)

%%time
snap.pp.knn(data)
snap.tl.leiden(data)

snap.pl.umap(data, color='leiden', interactive=False, height=500)

snap.pl.umap(data, color='leiden', show=False, out_file="umap.pdf", height=500)
snap.pl.umap(data, color='leiden', show=False, out_file="umap.html", height=500)

%%time
gene_matrix = snap.pp.make_gene_matrix(data, snap.genome.hg38)
gene_matrix

import scanpy as sc
pip install magic
pip install --user git+https://github.com/KrishnaswamyLab/MAGIC.git#subdirectory=python
sc.pp.filter_genes(gene_matrix, min_cells= 5)
sc.pp.normalize_total(gene_matrix)
sc.pp.log1p(gene_matrix)

%%time
sc.external.pp.magic(gene_matrix, solver="approximate")

%%time
sc.external.pp.magic(gene_matrix, solver="approximate")

# Copy over UMAP embedding
gene_matrix.obsm["X_umap"] = data.obsm["X_umap"]
marker_genes = ['MS4A1', 'CD3D', 'LEF1', 'NKG7', 'TREM1', 'LYZ', 'PPBP']
sc.pl.umap(gene_matrix, use_raw=False, color=["leiden"] + marker_genes)
```
