---
title: "Bulk RNA-seq Analysis of Skin Innate Lymphoid Cells: Comparison Between Wild-Type and Il18KO Mice"
permalink: /posts/2015-08-14-blog-post-27/
date: 2023-07-18
excerpt: 'This study employs bulk RNA-seq to compare the transcriptional profiles of skin ILCs from wild-type and Il18KO mice, revealing the impact of IL-18 signaling on immune-related pathways and skin ILC function. <br/>
<img src="/images/Skin_IL18/Il18_JAK.png" style="width:500px; height:500px;">'


tags:
  - Bioinformatics
---

Background
======
Innate lymphoid cells (ILCs) are a group of immune cells that play crucial roles in the regulation of immune responses, particularly at barrier sites such as the skin and intestines. These cells are classified into three major subsets based on their cytokine production profiles and functional roles: ILC1s, ILC2s, and ILC3s. Skin ILCs, a subset of ILCs, contribute to the maintenance of skin homeostasis, defense against infections, and modulation of inflammatory responses. Recent studies suggest that ILCs are highly influenced by local cytokine environments, which regulate their differentiation and function.<br/><br/>

Interleukin 18 (IL-18), a potent pro-inflammatory cytokine, is critical for the activation and function of ILCs in several tissues. Its receptor, IL-18R1, is expressed on various immune cell types, including ILCs, and has been implicated in modulating inflammatory responses. However, the specific role of IL-18 signaling in skin ILCs remains unclear. Understanding how IL-18 influences the homeostasis and activation of skin ILCs is important for elucidating its potential as a therapeutic target in inflammatory skin diseases such as psoriasis, atopic dermatitis, and eczema.<br/><br/>

To investigate the role of IL-18 in skin ILCs, we have utilized mice with targeted knockout mutations of both IL-18 (Il18KO) and Rag2 (Rag2KO). The Rag2 knockout renders these mice immunodeficient, lacking functional T and B cells, which allows for a focused study on innate immune components, particularly ILCs. The Il18KO model helps dissect the contribution of IL-18 signaling in these cells by eliminating its effects. By integrating these two genetic models, we aim to better understand the molecular mechanisms underlying IL-18's role in regulating skin ILCs in vivo.<br/><br/>




Result
======
**WT vs Il18KO, JAK-STAT** <br/> <img src="/images/Skin_IL18/Il18_JAK.png"><br/><br/>

**WT vs Il18KO, TF** <br/> <img src="/images/Skin_IL18/Il18_TF.png"><br/><br/>

**WT vs Rag2KO** <br/> <img src="/images/Skin_IL18/Rag2.png"><br/><br/>


Method
======
## Animal Models
Il18KO and Rag2KO mice were used to explore the role of IL-18 in skin ILCs. Il18KO mice were bred to eliminate the IL-18 cytokine, while Rag2KO mice, which lack functional T and B lymphocytes, were utilized to provide an immunodeficient background. The double knockout (Il18KO Rag2KO) model was generated to specifically investigate the effects of IL-18 deficiency in the absence of adaptive immune cells. Control groups consisted of age- and sex-matched wild-type (WT) C57BL/6J mice.<br/>

## Animal Models and Sample Information
Skin innate lymphoid cells (ILCs) were isolated from mice with various genotypes to investigate the role of IL-18 in skin ILC function. The following mouse strains were used in this study:<br/>

W1, W2: Wild-type (WT) C57BL/6J mice, serving as the control group for skin ILCs.<br/>

R1, R2: Rag2KO mice, which lack T and B lymphocytes, were used to examine innate immune responses in the absence of adaptive immunity.<br/>

B181, B182: Il18KO mice, where the IL-18 gene is knocked out, were used to assess the effect of IL-18 deficiency on skin ILCs.<br/>

## Skin ILC Isolation
Skin tissues were harvested from experimental and control mice. Epidermal sheets were obtained through enzymatic digestion with collagenase D and DNase I. The dermal layer was disrupted, and single-cell suspensions were prepared. ILCs were isolated by immunomagnetic sorting using a lineage depletion cocktail (CD3e, CD5, CD19, B220, CD11b, CD11c, Gr-1) to remove contaminating non-ILC populations. Enriched ILCs were further separated by flow cytometry based on surface markers Cd90 and Cd127, which are characteristic of ILC subsets.<br/>

## Bulk RNA Sequencing (RNA-seq)
Total RNA was extracted from purified skin ILCs using the RNeasy Plus kit (Qiagen). RNA integrity and quality were assessed, and only high-quality RNA samples were used for sequencing. Library preparation was conducted using the mRNA Library Prep Kit, which includes polyA selection and cDNA synthesis. Sequencing was performed to generate high-quality paired-end reads.<br/>

## Data Analysis
The raw sequencing data were processed and analyzed using standard RNA-seq pipelines. After quality trimming of reads, the data were aligned to the mouse reference genome (GRCm39) using STAR. Gene expression levels were quantified with featureCounts, and differential expression was analyzed using DESeq2. Functional enrichment analysis was performed with DAVID and Gene Set Enrichment Analysis (GSEA) to identify biological processes and pathways that are differentially regulated in skin ILCs with and without IL-18 signaling. Results were visualized using heatmaps, volcano plots, and principal component analysis (PCA) to compare gene expression profiles across experimental conditions.<br/>

## Conclusion
This study aims to dissect the role of IL-18 in regulating skin ILCs using Il18KO Rag2KO mice. By utilizing RNA-seq, we will generate comprehensive transcriptomic profiles of skin ILCs and identify pathways that are influenced by IL-18 signaling. This research has the potential to reveal new insights into the role of IL-18 in skin immunity and inflammatory diseases, providing a foundation for future therapeutic strategies targeting IL-18 in skin disorders.

```python
#!/bin/bash

# 2.mapping
cd './fastq'
if [ ! -d ../mapping ]
then
	mkdir ../mapping
fi

ls *_1.fastq.gz |while read id
do
    headname=${id%_1.fastq.gz} 
    hisat2 -p 70 --dta -x /data/labShare/Sequencing/GENOME/hisat2/grcm38_tran/genome_tran -1 ${headname}_1.fastq.gz -2 ${headname}_2.fastq.gz -S ../mapping/${headname}.sam
done



# 3.SamToBam & filter low-quality mapping
cd '../mapping'
ls *sam |while read id
do
    samtools view -@ 70 -b -F 4 -S ${id} -o ./${id%.sam}.bam
    samtools sort -@ 70 ./${id%.sam}.bam -o ./${id%.sam}.sorted.bam
	rm ${id}
	samtools index ./${id%.sam}.sorted.bam
done

# 4.counting
if [ ! -d ../count ]
then
	mkdir ../count
fi

```
